src/charm.py

```diff
@@ -160,8 +161,9 @@ def _update_layer(self) -> None:
             try:
                 self.logger.info("Pebble plan updated with new configuration, replaning")
                 self.container.replan()
-            except ChangeError:
-                raise ErrorWithStatus("Failed to replan", BlockedStatus)
+            except ChangeError as e:
+                self.logger.exception(f"Failed to replan, with error: {e}")
+                raise e
```

Comment:
```suggestion
            except ChangeError:
                self.logger.error(f"Failed to replan")
                raise
```

Reply:
We should include {e} in the log message. It will help in debugging.

Reply:
no
it will be a part of the exception

Reply:
```
try:
    1/0
except ZeroDivisionError:
    print("divide 0")
    raise
divide 0
Traceback (most recent call last):
  File "<input>", line 2, in <module>
ZeroDivisionError: division by zero

```

Reply:
The exception that is displayed in `juju status` message is too short (is there a way to make it longer?) and cuts off important details. If we add {e} to log message we will see it in `juju debug-log`. Otherwise `juju debug-log` will only show ""Failed to replan" which is kinda useless.

Reply:
there is a `raise` at the end, you will get all the traceback in the log

Reply:
Got it.

Reply:
new update on the code snippet, idea is
```python
try: 
    1/0
except ZeroDivisionError as e:
    raise ZeroDivisionError("Do not do it") from e
```

```
Traceback (most recent call last):
  File "<input>", line 2, in <module>
ZeroDivisionError: division by zero
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "<input>", line 4, in <module>
ZeroDivisionError: Do not do it
```

Reply:
then juju error on the juju status will be crystal clear while we keep all the traceback from the original issue in the log

Reply:
Sounds perfect.

Reply:
@beliaev-maksim `raise ChangeError("Failed to replan") from e` won't work, it has certain arguments that would be very hard for the charm code to determine. I think this applies to any exception that we raise. For instance, `ChangeError` needs `err: str, change: 'Change'`. I think it would be much easier to just log the error message and raise the exception.